src/sysusers/sysusers.c

```diff
@@ -352,6 +353,20 @@ static int sync_rights(FILE *from, FILE *to) {
         return 0;
 }
 
+static int restore_from_temporary(char **temp_path, const char *dest_path) {
+        int r = 0;
+        if (rename(*temp_path, dest_path) < 0)
+                return -errno;
+
+#ifdef SMACK_RUN_LABEL
+        r = mac_smack_apply(dest_path, SMACK_ATTR_ACCESS, SMACK_FLOOR_LABEL);
+        if (r < 0)
+                return r;
+#endif
+        *temp_path = mfree(*temp_path);
```

Comment:
I don't like this implicit pointer freeing, especially because the function name does not give a hint about this side effect. I'd prefer to stick with the old logic and call this function `rename_and_apply_smack()` or something.